Sources/SwiftPhoenixClient/Socket.swift

```diff
@@ -260,6 +260,7 @@ public class Socket: PhoenixTransportDelegate {
                                                paramsClosure: self.paramsClosure,
                                                vsn: vsn)
 
+    self.connection?.disconnect(code: CloseCode.normal.rawValue, reason: "connect called")
```

Comment:
The `connect` call is already checking that the socket is not connected before attempting to connect again.

```swift
// Do not attempt to reconnect if the socket is currently connected
guard !isConnected else { return }
```

So in theory, this should _never_ be used. Looking at the JS client (that you linked to in your PR), they only check if the connection exists, not if its been fully connected. So Maybe a better approach here is to update 

```
- guard !isConnected else { return }
+ guard self.connection == nil else { return } 
```

Reply:
> So in theory, this should never be used. 

Where I encountered issues with only `guard !isConnected else { return }` is calling `socket.connect()` from both `onAppear` and `onChange(of: scenePhase)` when scenePhase is `active` in a SwiftUI app. On occasion, the socket would be in the `connecting` state from the first call to `connect()` and pass the existing guard.

We've since removed the call to connect in `onAppear` to avoid this race condition entirely. 
One concern about `guard self.connection == nil else { return } ` is if the existing connection's state is `closing` or `closed`, I would probably expect a new connection to still be opened rather than be left with the existing closed one.

Reply:
Then I'd advocate for adding an `isConnecting` var and checking that as well

```swift
public var isConnecting: Bool {
    return self.connectionState == .connecting
  }


public func connect() {
    // Do not attempt to reconnect if the socket is currently connected
    guard !isConnecting && !isConnected else { return }
   ...
}

```